src/react/hooks/useMutation.ts

```diff
@@ -86,6 +86,14 @@ export function useMutation<
           ? new ApolloError({ graphQLErrors: errors })
           : void 0;
 
+      const errorPolicy = clientOptions.errorPolicy;
+
+      const onError = executeOptions.onError || ref.current.options?.onError
+
+      if (error && errorPolicy === 'all' && onError) {
```

Comment:
```suggestion
      if (error && onError) {
```

I believe some of the internals of Apollo Client (`QueryManager` in particular) will remove the `error` property from the result for you when `errorPolicy` is set to `ignore`, so the explicit check is not needed. Check to see if you're able to remove this check and maintain passing tests.

Reply:
Confirmed that removing this works as expected 🎉 . I'm going to go ahead and apply it :)